.github/workflows/main.yaml

```diff
@@ -0,0 +1,45 @@
+name: build
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch: # allows you to trigger manually
+
+jobs:
+  build:
+    if: "!contains(github.event.head_commit.message, 'skip ci')"
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          python -m pip install  -r requirements.txt
+
+      - name: Build site
+        run: |
+         cd site
+         nikola build
+
+      - name: Check links
+        run: |
+         cd site
+         nikola check --check-links --check-files --find-sources --remote --verbose
+      - name: Deploy to GitHub pages
+        if: github.ref == 'refs/heads/master'
```

Comment:
Does this line skip this step if it is just a PR to `master`?  That is, does this step only run if it is a push to `master`?

Reply:
> That is, does this step only run if it is a push to master? 

Yes